designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+features' is here broadly defined to mean syntactic changes that are one-line
+updates, such as replacing `<-` with `=` or renaming a variable/function.
+
+# Motivation
+[motivation]: #motivation
+
+Any large project accumulates many changes over time. While backwards
+compatibility is a strength of Stan, there are many instances where a small, yet
+breaking, change can dramatically improve code quality or maintainability in the
+long term. Some examples of this include adding keywords or removing outdated
+syntax which complicate parsing. Many of these features are _already_ marked as
+deprecated, but have been left for major version change to actually remove. This
+Stan 3.0 version seems to consistently be just over the horizon, and as a result
+some features, like the above-mentioned `<-` operator, have been deprecated for
+5+ years. Having the ability to remove or correct these tiny features allows the
+growth of Stan as a language to continue without an ever-increasing amount of
+difficulty.
+
+This is not a theoretical problem. Consider [this
+issue](https://github.com/stan-dev/stanc3/issues/953) with the parser or [this
+discussion](https://discourse.mc-stan.org/t/list-of-stanc3-new-reserved-keywords/12948)
+of new keywords.
+
+These are issues that stem from either the inability to make minor changes to
+the language, or from the current lack of process to introduce these changes.
+The solution to both of them is rather simple from a user perspective - the new
+`array[]` syntax can be automatically produced by our pretty-printer, and it is
+easy to rename a variable, but this must be clearly and consistently
+communicated to our users.
+
+Finally, this process provides an opportunity for feedback on changes. The
+deprecation period prior to removal is not only to let users know of our
+intentions, but also to let them tell us if our intentions need to be rethought.
+By communicating the changes and putting deadlines on them, we create a sort of
+community review period during which we can recieve feedback and course-correct
+if we have made a poor choice.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Deprecated minor features may be removed from Stan during minor version changes.
+Here, 'minor change' is restricted to be changes which can be done automatically
+through the use of the stanc3 canonicalizer, or a user-driven find and replace
+operation. By their very nature, minor changes are primarily syntactic. This is
+in contrast to 'major changes', which are by their nature semantic, and require
+human-in-the-loop reasoning to accommodate. Major changes, such the interface
+changes to the differential equation solvers, may be announced via deprecation
+warnings in minor versions, but will not be implemented/removed prior to a
+version 3.0.0.
+
+A minor feature must be deprecated for a minimum of 3 minor versions
+(approximately 1 year on the current release schedule) before removal. The exact
```

Comment:
I think we should just make this unequivocal and then argue exceptions as usual on a case-by-case basis.  Otherwise, I fear every single minor feature could potentially lead to a discussion of how long its deprecation period should be.